modules/Core/includes/endpoints/ServerInfoEndpoint.php

```diff
@@ -81,6 +81,24 @@ public function execute(Nameless2API $api): void {
             $api->throwError(CoreApiErrors::ERROR_UNABLE_TO_UPDATE_SERVER_INFO, $e->getMessage(), 500);
         }
 
+        // Server query
+        $query_type = Util::getSetting('external_query');
+        if ($query_type == QueryType::PLUGIN) {
+            $players_list = [];
+            foreach ($_POST['players'] as $uuid => $player) {
+                $players_list[] = ['id' => $uuid, 'name' => $player['name']];
+            }
+
+            $cache = new Cache(['name' => 'nameless', 'extension' => '.cache', 'path' => ROOT_PATH . '/cache/']);
+            $cache->setCache('latest_query');
+            $cache->store($server_id, [
+                'player_count' => count($_POST['players']),
+                'player_count_max' => $_POST['max_players'],
+                'player_list' => $players_list,
+                'motd' => $_POST['motd']
+            ], intval($_POST['interval_seconds'] ?? 10) * 2);
```

Comment:
Will we have to update the call to `validateParams` on line 13 to check for these as well?